core/domain/state_domain.py

```diff
@@ -2177,39 +2328,35 @@ def convert_html_fields_in_state(cls, state_dict, conversion_fn):
-            interaction_feedback_html = state_dict[
-                'interaction']['default_outcome']['feedback']['html']
-            state_dict['interaction']['default_outcome']['feedback'][
-                'html'] = conversion_fn(interaction_feedback_html)
+            state_dict['interaction']['default_outcome'] = (
+                Outcome.convert_html_in_outcome(
+                    state_dict['interaction']['default_outcome'],
+                    conversion_fn))
 
         for answer_group_index, answer_group in enumerate(
                 state_dict['interaction']['answer_groups']):
-            answer_group_html = answer_group['outcome']['feedback']['html']
-            state_dict['interaction']['answer_groups'][
-                answer_group_index]['outcome']['feedback']['html'] = (
-                    conversion_fn(answer_group_html))
-            if state_dict['interaction']['id'] == 'ItemSelectionInput':
-                for rule_spec_index, rule_spec in enumerate(
-                        answer_group['rule_specs']):
-                    for x_index, x in enumerate(rule_spec['inputs']['x']):
-                        state_dict['interaction']['answer_groups'][
-                            answer_group_index]['rule_specs'][
-                                rule_spec_index]['inputs']['x'][x_index] = (
-                                    conversion_fn(x))
-        for hint_index, hint in enumerate(
-                state_dict['interaction']['hints']):
-            hint_html = hint['hint_content']['html']
-            state_dict['interaction']['hints'][hint_index][
-                'hint_content']['html'] = conversion_fn(hint_html)
+            state_dict['interaction']['answer_groups'][answer_group_index] = (
+                AnswerGroup.convert_html_in_answer_group(
+                    answer_group, conversion_fn))
+
+        if 'written_translations' in state_dict.keys():
+            state_dict['written_translations'] = (
+                WrittenTranslations.
+                convert_html_in_written_translations(
+                    state_dict['written_translations'], conversion_fn))
+
+        for hint_index, hint in enumerate(state_dict['interaction']['hints']):
+            state_dict['interaction']['hints'][hint_index] = (
+                Hint.convert_html_in_hint(hint, conversion_fn))
 
         if state_dict['interaction']['solution']:
-            solution_html = state_dict[
-                'interaction']['solution']['explanation']['html']
-            state_dict['interaction']['solution']['explanation']['html'] = (
-                conversion_fn(solution_html))
+            state_dict['interaction']['solution'] = (
+                Solution.convert_html_in_solution(
+                    state_dict['interaction']['solution'], conversion_fn))
 
         if state_dict['interaction']['id'] in (
-                'ItemSelectionInput', 'MultipleChoiceInput'):
+                'ItemSelectionInput', 'MultipleChoiceInput',
```

Comment:
These should be found programmatically, not hardcoded. Remember what was mentioned in earlier reviews -- no direct references to specific interactions in the core code.

Reply:
We need this because this is not related to the rule_specs. This refers to the customization_args and we don't have a declarative list for that as of now(This approach was already being used in the `convert_html_fields_in_state`).

Reply:
Then a declarative list for that should be created using similar principles (though it doesn't look like it needs to be done in this PR).